source/memtrace.c

```diff
@@ -134,7 +134,7 @@ static void s_alloc_tracer_track(struct alloc_tracer *tracer, void *ptr, size_t
         /* capture stack frames, skip 2 for this function and the allocation vtable function */
         AWS_VARIABLE_LENGTH_ARRAY(void *, stack_frames, (FRAMES_TO_SKIP + tracer->frames_per_stack));
         size_t stack_depth = aws_backtrace(stack_frames, FRAMES_TO_SKIP + tracer->frames_per_stack);
-        if (stack_depth >= FRAMES_TO_SKIP) {
+        if (stack_depth > 0) {
```

Comment:
we talked, and maybe there's some other code that crashes if this is 0?
We should either:
- fix that other code, to allow stack_depth of 0
- or FATAL_ASSERT assert here that stack_depth > 0

Reply:
discussed offline. lets assert for now that we have a stack trace, since the rest of the memtraces assumes that allocations have a stack trace.